### PR TITLE
Implemented policy compliance checking function for Android plugin

### PR DESCRIPTION
**Pull request by milanperera**

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java [hunk 1]
```diff
@@ -35,36 +39,49 @@
...
+        JsonElement jsonElement = new JsonParser().parse((String) compliancePayload);
+        JsonArray jsonArray = jsonElement.getAsJsonArray();
+        Gson gson = new Gson();
+        ComplianceFeature complianceFeature;
 
-	@Override
-	public String getType() {
-		return DeviceManagementConstants.MobileDeviceTypes.MOBILE_DEVICE_TYPE_ANDROID;
-	}
+        for (int i = 0; i < jsonArray.size(); i++) {
```
Comment (prabathabey) [inline components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java]: If having access to indices is not a must, you can use a properly scoped iterator based for loop.

File: components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java [hunk 2]
```diff
@@ -35,36 +39,49 @@
...
-		for (ComplianceFeature cf : complianceFeatures) {
-			if(!cf.isCompliance()){
-				complianceData.setStatus(false);
-				break;
-			}
-		}
-		return complianceData;
-	}
+        // Parsing json string to get compliance features.
+        JsonElement jsonElement = new JsonParser().parse((String) compliancePayload);
```
Comment (nirmal070125) [inline components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java]: Use instaceof before casting.

File: components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java (hunk 1 above)
Comment (milanperera) [inline components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java]: Fixed in PR [https://github.com/wso2/carbon-device-mgt-plugins/pull/12]

File: components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java (hunk 2 above)
Comment (milanperera) [inline components/device-mgt/org.wso2.carbon.device.mgt.mobile.impl/src/main/java/org/wso2/carbon/device/mgt/mobile/impl/android/AndroidPolicyMonitoringService.java]: Added in PR [https://github.com/wso2/carbon-device-mgt-plugins/pull/12]

**Pull request merged**